### PR TITLE
Image Upload: rename `ProductImagesService` & `ProductImagesProvider`

### PR DESCRIPTION
**Pull request by jaclync**
Rename `ProductImagesService` & rename `ProductImagesProvider` for #1713 

This is a followup task to rename `ProductImagesService` and `ProductImagesProvider` for better readability.

## Changes

- Renamed `ProductImagesService` to `ProductImageActionHandler` & variable names
- Renamed `ProductImagesProvider` to `ProductUIImageLoader`, `DefaultProductImagesProvider` to `DefaultProductUIImageLoader`, and variable names

## Testing

Since this PR is just about renaming, I think CI suffices. However, feel free to smoke test the app, adding some Product images!

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductUIImageLoaderTests.swift [hunk 1]
```diff
@@ -27,7 +27,7 @@ final class ProductImagesProviderTests: XCTestCase {
 
     func testRequestingImageWithRemoteProductImage() {
         let mockPHAssetImageLoader = MockPHAssetImageLoader(imagesByAsset: [:])
-        let imagesProvider = DefaultProductImagesProvider(imageService: imageService, phAssetImageLoader: mockPHAssetImageLoader)
+        let imagesProvider = DefaultProductUIImageLoader(imageService: imageService, phAssetImageLoader: mockPHAssetImageLoader)
```
Comment (shiki) [inline WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductUIImageLoaderTests.swift]: We missed this one. 🙂 

```suggestion
        let imageLoader = DefaultProductUIImageLoader(imageService: imageService, phAssetImageLoader: mockPHAssetImageLoader)
```

File: WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductUIImageLoaderTests.swift [hunk 2]
```diff
@@ -46,7 +46,7 @@ final class ProductImagesProviderTests: XCTestCase {
     func testRequestingImageWithPHAsset() {
         let asset = PHAsset()
         let mockPHAssetImageLoader = MockPHAssetImageLoader(imagesByAsset: [asset: testImage])
-        let imagesProvider = DefaultProductImagesProvider(imageService: imageService, phAssetImageLoader: mockPHAssetImageLoader)
+        let imagesProvider = DefaultProductUIImageLoader(imageService: imageService, phAssetImageLoader: mockPHAssetImageLoader)
```
Comment (shiki) [inline WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductUIImageLoaderTests.swift]: We missed this one. 🙂 

```suggestion
        let imageLoader = DefaultProductUIImageLoader(imageService: imageService, phAssetImageLoader: mockPHAssetImageLoader)
```

Comment (jaclync) [reply WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductUIImageLoaderTests.swift]: updated in: https://github.com/woocommerce/woocommerce-ios/pull/1914/commits/bef63d1cf1e24112f0cbf95be40b67d224a16658

File: WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductUIImageLoaderTests.swift (hunk 1 above)
Comment (jaclync) [reply WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductUIImageLoaderTests.swift]: yup! updated in: https://github.com/woocommerce/woocommerce-ios/pull/1914/commits/bef63d1cf1e24112f0cbf95be40b67d224a16658

**Comment by jaclync** [comment]
thanks for catching the variables in tests, @shiki ! merging after updating in https://github.com/woocommerce/woocommerce-ios/pull/1914/commits/bef63d1cf1e24112f0cbf95be40b67d224a16658

**Pull request merged**